pycode/epidemiology/epidata/getDIVIData.py

```diff
@@ -47,380 +47,40 @@
-    - April 24, 2020.: Rename column 'kreis' to 'gemeindeschluessel'
-    - April 25, 2020.: Add id bundesland, which is extracted from the given column "gemeindeschluessel"
-    - April 24-27, 2020.: Add date as a column
-    - April 29, 2020.: Empty column has to be removed
-    - Before March 31, 2021 rename faelle_covid_aktuell_beatmet' to  'faelle_covid_aktuell_invasiv_beatmet'
-
-    @param df A pandas data frame
-    @param date_of_data The date for the data stored in df
-    @return changed pandas dataframe
-    """
-
-    # rename column names due to change in naming of files from March 31, 2021 onwards
-    df.rename(columns={
-        'faelle_covid_aktuell_beatmet': 'faelle_covid_aktuell_invasiv_beatmet'}, inplace=True)
-
-    # from October 29, 2021 'daten_stand' was renamed into 'date'
-    df.rename(columns={
-        'daten_stand': dd.EngEng['date']}, inplace=True)
-    df.rename(columns={
-        'date': dd.EngEng['date']}, inplace=True)
-
-    # rename column 'kreis' of first date to match data of following days
-    if date_of_data == date(2020, 4, 24):
-        df.rename({'kreis': 'gemeindeschluessel'}, axis=1, inplace=True)
-        # Add empty column ICU_ventilated
-        df.insert(loc=len(df.columns),
-                  column='faelle_covid_aktuell_invasiv_beatmet', value='')
-
-    # remove empty column
-    if date_of_data == date(2020, 4, 29):
-        df.drop(columns='Unnamed: 0', inplace=True)
-
-    # add dates for data until 27.4.
-    if date_of_data <= date(2020, 4, 27):
-        date_str = date_of_data.strftime("%Y-%m-%d") + " 09:15:00"
-        df.insert(loc=len(df.columns),
-                  column=dd.EngEng['date'], value=date_str)
-
-    # add 'bundesland' for data from 25.4.
-    if date_of_data == date(2020, 4, 25):
-        df.insert(loc=0, column='bundesland',
-                  value=df['gemeindeschluessel'].floordiv(1000))
-        df.insert(loc=len(df.columns),
-                  column='faelle_covid_aktuell_invasiv_beatmet', value='')
-
-    return df
-
-
-def call_call_url(url_prefix, call_number):
-    """!Calls, if possible a url and breaks if it is not possible to call the url,
-    e.g., due to a not working internet connection
-    or returns an empty dataframe if url is not the correct one.
-
-    @param url_prefix Date specific part of the url
-    @param call_number Irregular number which is needed for the url
-
-    @return pandas dataframe which is either empty or contains requested data
-    """
-
-    call_url = "https://www.divi.de/divi-intensivregister-tagesreport-archiv-csv" + "/viewdocument/" \
-               + str(call_number) + "/divi-intensivregister-" + url_prefix
-
-    # empty data frame
-    df = pandas.DataFrame()
-
-    try:
-        df = pandas.read_csv(call_url)
-    except:
-        pass
+    startdelta = (start_date - dd.defaultDict['start_date']).days
+    enddelta = (dd.defaultDict['end_date'] - end_date).days
+    for i in range(startdelta):
```

Comment:
I don't understand this functionality at all and it may be much too expensive. 

1.) Why do you consider dd.defaultDict['start_date']? dd.defaultDict['start_date'] should only be considered as a standard argument as long as no other start_date is given to the function.
2.) Wouldn't you append the df_new to df instead of overwritting it?
3.) Please never do df.Date or similar. Use the dd.EngEng dictionary to avoid wrong access if columns are changed globally !
4.) If you just want to extract a subframe, why not going for
df = df[(df[dd.EngEng['date']]< x) & (... > y)] (or similar)